File: festival-web/src/main/java/cn/bdqfork/web/annotation/PermitAllowed.java
package cn.bdqfork.web.annotation;

import cn.bdqfork.web.constant.LogicType;

import java.lang.annotation.*;

/**
 * 表示哪些权限可以访问api
 *
 * @author bdq
 * @since 2020/1/27
 */
@Auth
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface PermitAllowed {
    String[] value();

    LogicType logic() default LogicType.AND;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/annotation/PermitAllowed.kt
package cn.bdqfork.web.annotation

import cn.bdqfork.web.constant.LogicType

/**
 * 表示哪些权限可以访问api
 *
 * @author bdq
 * @since 2020/1/27
 */
@Auth
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.FUNCTION)
annotation class PermitAllowed(vararg val value: String, val logic: LogicType = LogicType.AND)